internal/hcsoci/resources_lcow.go

```diff
@@ -124,5 +134,43 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, res
 		}
 	}
 
+	if coi.HostingSystem != nil && coi.Spec.Windows != nil {
+		addGpuVhd := false
```

Comment:
You have to track this at the uvm level. To containers should not add this vhd twice. So we cant add it in the create path and store that as a local

Reply:
why not? With abdul's changes, it can be ref counted each time a container attempts to add it and then removed from the uvm when the last container with gpu support is cleaned up.

Reply:
I'm cool with that!